flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskManagerJobMetricGroup.java

```diff
@@ -38,6 +39,7 @@
  * <p>Contains extra logic for adding Tasks ({@link TaskMetricGroup}).
  */
 @Internal
+@NotThreadSafe
```

Comment:
I thought we didn't want to change this?

Reply:
You're right, I only restored thread safety for `TaskManagerMetricGroup`.